.github/workflows/ci.yaml

```diff
@@ -35,9 +35,64 @@ jobs:
   run-reuse-workflow:
     uses: opalmedapps/.github/.github/workflows/reuse.yaml@main
 
+  test-mocha:
+    runs-on: ubuntu-latest
+    steps:
+      # Setup
+      - uses: actions/checkout@v4.2.2
+        with:
+          persist-credentials: false
+      - name: Install dependencies
+        run: npm ci
+      - name: Print mocha version
+        run: npx mocha --version
+
+      # Run tests
+      - name: Run mocha
+        run: npx mocha --reporter json --reporter-option output=mocha-report.json './src/**/*.test.*' './listener/**/*.test.*' './legacy-registration/**/*.test.*'
+
+      # Format the output for GitHub
+      - name: Report test results
+        uses: dorny/test-reporter@v2.1.0
+        # Run this step even if the previous step failed
+        if: ${{ !cancelled() }}
+        with:
+          name: Mocha Tests
+          path: mocha-report.json
+          reporter: mocha-json
+
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      # Setup
+      - uses: actions/checkout@v4.2.2
+        with:
+          persist-credentials: false
+      - name: Install dependencies
+        run: npm ci
+
+      # Run coverage evaluation
+      - name: Run nyc / cobertura
```

Comment:
Is there a way to combine test and coverage since both require a test run?

Reply:
I've managed to combine them and it seems to work fine; the job summaries follow each other under `test-and-coverage summary`.

See here for a successful run: https://github.com/opalmedapps/opal-listener/actions/runs/15401384923?pr=430
And here for an error run (failing test): https://github.com/opalmedapps/opal-listener/actions/runs/15401619080?pr=430

One thing to consider is that it'll be harder to distinguish failure due to tests vs failure due to coverage later on. For now though, coverage is set to always succeed.

Reply:
Thanks! I think for now that should be sufficient. Unless you think that it is better to have them as separate jobs.

Reply:
I agree, I think it's fine for now.